From 858b857a91bd62b7a04f9603e78911765485d1dc Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Wed, 12 Feb 2014 19:08:42 -0800
Subject: [PATCH] usb: gadget: atmel_usba_udc: remove leftover kfree()s that
 cause oops

These two kfree() calls were not included in a patch back-porting the
device tree changes. The DT patch changed to use devm_kzalloc() instead.
Memory allocated with this function are automatically freed so leaving
these in causes a double-free condition, eventually crashing the kernel.

Bug: 4889
---
 drivers/usb/gadget/atmel_usba_udc.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/usb/gadget/atmel_usba_udc.c b/drivers/usb/gadget/atmel_usba_udc.c
index 013f410..5e7886a 100644
--- a/drivers/usb/gadget/atmel_usba_udc.c
+++ b/drivers/usb/gadget/atmel_usba_udc.c
@@ -2244,7 +2244,6 @@ err_add_udc:
 err_device_add:
 	free_irq(irq, udc);
 err_request_irq:
-	kfree(usba_ep);
 err_alloc_ep:
 	iounmap(udc->fifo);
 err_map_fifo:
@@ -2282,7 +2281,6 @@ static int __exit usba_udc_remove(struct platform_device *pdev)
 	}
 
 	free_irq(udc->irq, udc);
-	kfree(usba_ep);
 	iounmap(udc->fifo);
 	iounmap(udc->regs);
 	clk_put(udc->hclk);
-- 
1.7.9.5

